openai/openai.go

```diff
@@ -148,7 +149,33 @@ func toChunk(id string, r api.ChatResponse) ChatCompletionChunk {
 func fromRequest(r ChatCompletionRequest) api.ChatRequest {
 	var messages []api.Message
 	for _, msg := range r.Messages {
-		messages = append(messages, api.Message{Role: msg.Role, Content: msg.Content})
+		switch content := msg.Content.(type) {
+		case string:
+			messages = append(messages, api.Message{Role: msg.Role, Content: content})
+		case []any:
+			for _, c := range content {
+				if data, ok := c.(map[string]any); ok {
+					message := api.Message{Role: msg.Role}
+					switch data["type"] {
+					case "text":
+						if text, ok := data["text"].(string); ok {
+							message.Content = text
+						}
+					case "image_url":
+						if urlMap, ok := data["image_url"].(map[string]any); ok {
+							if url, ok := urlMap["url"].(string); ok {
+								if img, err := base64.StdEncoding.DecodeString(url); err == nil {
+									message.Images = append(message.Images, img)
+								}
+							}
+						}
+					}
+					if message.Content != "" || len(message.Images) > 0 {
```

Comment:
Is there a case where this `if` statement wouldn't be true?

Reply:
There's a weird edge case where an assistant message can be passed in without content if tool_calls or function_call is specified, but I can still take it out let me know
<img width="664" alt="Screenshot 2024-06-24 at 11 29 14 AM" src="https://github.com/ollama/ollama/assets/65097070/152cddc6-9340-422d-9e41-b036d3edd217">